pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -180,6 +187,18 @@ func (c *Check) Connect() (*sqlx.DB, error) {
 		c.tags = append(c.tags, fmt.Sprintf("host:%s", c.dbHostname), fmt.Sprintf("oracle_version:%s", c.dbVersion))
 	}
 
+	if c.filePath == "" {
+		r := db.QueryRow("SELECT SUBSTR(name, 1, 10) path FROM v$datafile WHERE rownum = 1")
+		var path string
+		err = r.Scan(&path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to query path: %w", err)
+		}
+		if path == "/rdsdbdata" {
+			c.isRDS = true
+		}
```

Comment:
Rather than doing this iteration, could we simply modify the query to filter on this path?

Reply:
I can change it if you insist, but the current implementation is more efficient because if we put the condition in the `WHERE` clause, the whole result set must be traversed on a non-RDS database to make sure the condition is not satisfied for every row. The current implementation always selects only a single row and evaluates the condition outside the database.

Reply:
I don't insist, that's valid 👍